centreon/src/Core/Dashboard/Infrastructure/API/FindMetricsTop/FindMetricsTopController.php

```diff
@@ -26,42 +26,27 @@
 use Centreon\Application\Controller\AbstractController;
 use Core\Dashboard\Application\UseCase\FindMetricsTop\FindMetricsTop;
 use Core\Dashboard\Application\UseCase\FindMetricsTop\FindMetricsTopRequest;
-use Symfony\Component\HttpFoundation\Request;
 use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\Attribute\MapQueryString;
 
 final class FindMetricsTopController extends AbstractController
 {
-    private const METRIC_NAME_PARAMETER = 'metric_name';
-
     /**
      * @param FindMetricsTop $useCase
      * @param FindMetricsTopPresenter $presenter
-     * @param Request $request
+     * @param FindMetricsTopRequest $request
      *
      * @return Response
      */
-    public function __invoke(FindMetricsTop $useCase, FindMetricsTopPresenter $presenter, Request $request): Response
-    {
+    public function __invoke(
+        FindMetricsTop $useCase,
+        FindMetricsTopPresenter $presenter,
+        #[MapQueryString] FindMetricsTopRequest $request
```

Comment:
Consider adding the validationFailedStatusCode here too

Reply:
Done